### PR TITLE
Resolt to first interface name if node name is unusable

### PR DESCRIPTION
**Pull request by jpnurmi**

### CODE REVIEW

**Comment by jpnurmi** [comment]
Many nodes don't either have a name at all, or just "/":
```bash
$ grep -r "<node" /usr/share/dbus-1/interfaces
/usr/share/dbus-1/interfaces/org.fedoraproject.Config.Printing.xml:<node name="/org/fedoraproject/Config/Printing"
/usr/share/dbus-1/interfaces/org.freedesktop.PackageKit.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.GeoClue2.Agent.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.DomainBrowser.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.ColorManager.Sensor.xml:<node name="/" xmlns:doc="https://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.Accounts.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.UPower.Device.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.PackageKit.Transaction.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.GeoClue2.xml:<node><interface name="org.freedesktop.GeoClue2.Location">
/usr/share/dbus-1/interfaces/org.freedesktop.UPower.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.EntryGroup.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.AddressResolver.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.ServiceTypeBrowser.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.ColorManager.Device.xml:<node name="/" xmlns:doc="https://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.Server.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.ServiceResolver.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.RealtimeKit1.xml:<node>
/usr/share/dbus-1/interfaces/org.gnome.Shell.Screencast.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.UPower.KbdBacklight.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.GeoClue2.Manager.xml:<node>
/usr/share/dbus-1/interfaces/org.gnome.ShellSearchProvider2.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Accounts.User.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.gnome.ShellSearchProvider.xml:<node>
/usr/share/dbus-1/interfaces/org.gnome.Shell.Screenshot.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.fwupd.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.RecordBrowser.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.GeoClue2.Client.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.ColorManager.Profile.xml:<node name="/" xmlns:doc="https://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.gnome.Shell.Introspect.xml:<node>
/usr/share/dbus-1/interfaces/org.gnome.Shell.PadOsd.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.bolt.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.ColorManager.xml:<node name="/" xmlns:doc="https://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.GeoClue2.Location.xml:<node>
/usr/share/dbus-1/interfaces/org.gnome.Shell.Extensions.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.ServiceBrowser.xml:<node>
/usr/share/dbus-1/interfaces/org.freedesktop.UPower.Wakeups.xml:<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.ColorHelper.xml:<node name="/" xmlns:doc="https://www.freedesktop.org/dbus/1.0/doc.dtd">
/usr/share/dbus-1/interfaces/org.freedesktop.Avahi.HostNameResolver.xml:<node>
```

For example, with `dart-dbus generate(-remote)-object /usr/share/dbus-1/interfaces/org.freedesktop.UPower.xml`:

### before:


```dart
class  extends DBusObject {
  // ...
}

```


```dart
class  extends DBusRemoteObject {
  (DBusClient client, String destination, {DBusObjectPath path = const DBusObjectPath.unchecked('/')}) : super(client, destination, path);
  // ...
}
```

### after:

```dart
class OrgFreedesktopUPowerDevice extends DBusObject {
  // ...
}
```

```dart
class OrgFreedesktopUPowerDevice extends DBusRemoteObject {
  OrgFreedesktopUPowerDevice(DBusClient client, String destination, {DBusObjectPath path = const DBusObjectPath.unchecked('/')}) : super(client, destination, path);
  // ...
}
```

**Comment by robert-ancell** [comment]
Thanks @jpnurmi! Can you please sign the [Canonical contributor license agreement](https://ubuntu.com/legal/contributors)? Apologies for that not being documented yet, I have a task to do that and set it up in the CI. Note that the form says you need a Launchpad account, but I don't think that's a requirement anymore.

**Comment by jpnurmi** [comment]
Hi Robert, I've signed the agreement. It was asking for a Canonical contact in the form, so I put your name. I hope that's ok. :)

File: bin/dart-dbus.dart [hunk 1]
```diff
@@ -644,15 +646,15 @@ String generateRemoteSignalSubscription(
 /// Converts a D-Bus path to a Dart class name. e.g. 'org.freedesktop.Notifications' -> 'OrgFreedesktopNotifications'.
 String pathToClassName(String path) {
   var className = '';
-  for (var element in path.split('/')) {
+  for (var element in path.split(RegExp('[/\.]'))) {
```
Comment (robert-ancell) [inline bin/dart-dbus.dart:649]: This made me notice the function is wrongly named - it was actually `interfaceToClassName` and now it's `interfaceOrPathToClassName`. But I'll follow that up in a following commit.

**Comment by robert-ancell** [comment]
Thanks @jpnurmi!

**Pull request merged**